asserts/snap_asserts.go

```diff
@@ -110,6 +117,8 @@ func (snapdcl *SnapDeclaration) Prerequisites() []*Ref {
 	}
 }
 
+var validAppCommand = regexp.MustCompile("^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$")
```

Comment:
Do we want to be as strict for toplevel commands? AIUI this will exclude commands like `pg_dump` (?).